### DISCUSSION TITLE
Missing code for tud_hid_get_report_cb and tud_hid_set_report_cb (undefined reference)

### DISCUSSION THREAD

**Discussion by matias-vago** [hathach/tinyusb#2241] [Q&A]
Hi, im trying to use the library to make a usb composite device (mouse and keyboard) with STM32F103c8t6 and ive been following the integration guide at https://github.com/hathach/tinyusb/discussions/633. The only issue ive had to compile that i cant seem to solve is i get error undefined reference to tud_hid_get_report_cb and tud_hid_set_report_cb in hid_device.c file (this file has a single function call to each one of these functiones). Looking through the code i notice that the function prototypes for these functiones are in hid_device.h, but the actual functions i cant seem to find. Is it posible they are missing? I made dummy function definitions in hid_device.c for each one just to see if that was the problem, and this way the code does compile.

**Comment by Rocky04** [comment]
Your application need to provide them as shown in the examples, [like here](https://github.com/hathach/tinyusb/blob/46f7cf4da2959b96d3af6b7f23d0d02ff7187f2f/examples/device/hid_boot_interface/src/main.c#L214). _They are not weak and so they must be present._
